### PR TITLE
feat(data-exploration): convert funnel correlation to data exploration

### PR DESCRIPTION
**Pull request by thmsobrmlr**
## Overview: Conversion of Funnel Correlation to Data Exploration

Conversion of funnel correlation to data exploration is a wrap!

I've stacked the relevant PRs on top of each other for easier review-ability, but for clicking around feel free to go straight to the last one https://github.com/PostHog/posthog/pull/14992.

Here's a brief summary of the PRs:
- https://github.com/PostHog/posthog/pull/14963: Some prep work. Getting rid of obsolete code and splitting out components into their own files.
- https://github.com/PostHog/posthog/pull/14970: Conversion of the "skew warning" 
- https://github.com/PostHog/posthog/pull/14971: Conversion of the feedback form
- https://github.com/PostHog/posthog/pull/14975: Separating out the usage tracking to it's own logic
- https://github.com/PostHog/posthog/pull/14977: Conversion of the funnel correlations table
- https://github.com/PostHog/posthog/pull/14982: Conversion of the funnel correlations details
- https://github.com/PostHog/posthog/pull/14992: Conversion of the funnel property correlations table

In broad terms, I've split out the correlations related code from `funnelLogic` into separate logics with a clear purpose:
- `funnelCorrelationLogic`: event and event property correlation
- `funnelPropertyCorrelationLogic`: property correlation
- `funnelCorrelationDetailsLogic`: the details popup
- `funnelCorrelationFeedbackLogic`: the feedback form
- `funnelCorrelationUsageLogic`: usage tracking with posthog.js

The components and logics then disambiguate between data exploration and legacy mode for everything that is pulled in from the funnel.

I'm going to give this one more review before merging in myself, but the whole thing is ready for review now.

---

## Problem

In order to convert the funnel correlation work to data exploration, the `<FunnelCorrelation />` component needed a bit of preparation. I'm going to stack other PRs converting the funnel correlations on top.

## Changes

This PR:
- removes the correlation analysis available feature check, after it became obsolete by https://github.com/PostHog/posthog/pull/13738
- moves the individual funnel correlation components to their own files, so that they ideally can be converted one-by-one
- re-factors the "correlation actions cell" into a unified component

## How did you test this code?

Manual testing

### CODE REVIEW

**Comment by thmsobrmlr** [comment]
I don't know why GitHub thinks this should need Django tests.

**Review by mariusandra — APPROVED** [review]
Everything worked fine when clicking around in the interface. Going through the code now. This chunk seems legit 👍

**Review by Twixes — APPROVED** [review]
Looks great to me

File: frontend/src/queries/nodes/InsightViz/InsightContainer.tsx [hunk 1]
```diff
@@ -30,12 +30,12 @@ import {
 import { PathCanvasLabel } from 'scenes/paths/PathsLabel'
 import { InsightLegend } from 'lib/components/InsightLegend/InsightLegend'
 import { InsightLegendButtonDataExploration } from 'lib/components/InsightLegend/InsightLegendButton'
-// import { FunnelCorrelation } from './views/Funnels/FunnelCorrelation'
 // import { AlertMessage } from 'lib/lemon-ui/AlertMessage'
```
[the remark this answers is not in the document]
Comment (thmsobrmlr) [reply frontend/src/queries/nodes/InsightViz/InsightContainer.tsx:33]: That'll go in again when I implement the `isUsingSessionAnalysis` flag.

**Pull request merged**